pyautomagic/src/calcQuality.py

```diff
@@ -0,0 +1,83 @@
+from typing import List
+import logging
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+def calcQuality(data: np.ndarray, bad_chans: List, overallThresh: float = 50, timeThresh: float = 25,
+                chanThresh: float = 25, apply_common_avg: bool = True):
+    """
+    Calculates four essential quality metrics of a data set and returns a dictionary of the quality metrics and thresholds/settings used
+
+    The four quality metrics:
+        Overall high amplitude data points
+        Timepoints of high variance across channels
+        Ratio of bad channels
+        Channels of high variance across time
+
+    Mean absolute voltage is also calculated.
+
+    :param Data: a channels x timepoints data array of EEG
+    :type Data: np.ndarray
+    :param bad_chans:  a list of the numbers of bad channels
+    :type bad_chans: list
+    :param overallThresh: overall threshold for rejection of ...
+    :type overallThresh: float
+    :param timeThresh: threshold for rejecting time segments
+    :type timeThresh: float
+    :param chanThresh: threshold for rejecting channels
+    :type chanThresh: float
+    :param apply_common_avg: if average referencing should be applied
+    :type apply_common_avg: bool
+    :return: quality_metrics: a dictionary of the quality metrics and thresholds/settings used
+        The four quality metrics:
+        Overall high amplitude data points
+        Timepoints of high variance across channels
+        Ratio of bad channels
+        Channels of high variance across time
+
+        Mean absolute voltage also calculated
+    :rtype: dict
+    """
+    # checking all of the default value types, if not what they should be, use default
+    if not isinstance(overallThresh, int) and not isinstance(overallThresh, float) or isinstance(overallThresh, bool) or (overallThresh <= 0):
+        overallThresh = 50
+        logger.log(30,'Invalid overallThresh value. Default of 50 used.')
+    if not isinstance(timeThresh, int) and not isinstance(timeThresh, float) or isinstance(timeThresh, bool) or (timeThresh <= 0):
+        timeThresh = 25
+        logger.log(30,'Invalid timeThresh value. Default of 25 used.')
+    if not isinstance(chanThresh, int) and not isinstance(chanThresh, float) or isinstance(chanThresh,bool) or (chanThresh <= 0):
+        chanThresh = 25
+        logger.log(30,'Invalid chanThresh value. Default of 25 used.')
+    if not isinstance(apply_common_avg, bool):
+        # ADD STATEMENT TO ALLOW TYPICAL OTHER BOOLEAN INDICATORS
+        apply_common_avg = True
+        logger.log(30,'Invalid apply_common_avg value. Average referencing used.')
+    # ADD CHECKS FOR DAT AND BAD_CHANS?
```

Comment:
Will you do the verification for dat and bad_chans? If not, you can erase this comment